contracts/handler/CompatibilityFallbackHandler.sol

```diff
@@ -169,6 +169,27 @@ contract CompatibilityFallbackHandler is TokenCallbackHandler, ISignatureValidat
      * @param requiredSignatures Amount of required valid signatures.
      */
     function checkNSignatures(bytes32 dataHash, bytes memory, bytes memory signatures, uint256 requiredSignatures) public view {
-        Safe(payable(_manager())).checkNSignatures(_msgSender(), dataHash, "", signatures, requiredSignatures);
+        Safe(payable(_manager())).checkNSignatures(_msgSender(), dataHash, signatures, requiredSignatures);
+    }
+
+    /**
+     * @notice Checks whether the signature provided is valid for the provided data and hash. Reverts otherwise.
+     * @dev Since the EIP-1271 does an external call, be mindful of reentrancy attacks.
+     *      The function was moved to the fallback handler as a part of
+     *      1.5.0 contract upgrade. It used to be a part of the Safe core contract, but
+     *      was replaced by the same function that removes the encoded bytes data as a parameter
+     *      and only accepts the data hash.
+     *      The data parameter (bytes) is not used since v1.5.0 as it is not required anymore. Prior to v1.5.0,
+     *      data parameter was used in contract signature validation flow using legacy EIP-1271.
+     *      Version v1.5.0, uses dataHash parameter instead of data with updated EIP-1271 implementation.
```

Comment:
I would either mention that the data parameter is ignored, or `require(keccak256(data) == dataHash)`.

Reply:
I added a comment to the variable name